### ISSUE TITLE
[A11Y] Breadcrumb separator should be marked as decorative

### ISSUE THREAD

**Issue by herbdool** [backdrop/backdrop-issues#6034]
A couple resources ([here](https://www.boia.org/blog/how-can-i-make-breadcrumbs-accessible-for-people-with-disabilities) and [here](https://www.aditus.io/patterns/breadcrumbs/)) recommend "decorative separators" by using either CSS or the aria-hidden="true" for the separator.

 Core currently just returns ` » ` so this means a screen reader will announce this as text.

Since in https://github.com/backdrop/backdrop-issues/issues/2351 there was some hesitancy of changing it to CSS since it might not work well with current theme implementations, I think we should instead wrap the separator like this:

```
<span class="breadcrumb-separator" aria-hidden="true"> » </span>
```

This would have the side benefit of allowing themes to hide the separator with just CSS and add their own. There is some risk some themes would need some minor fixing, but I think it's pretty low. I think that risk is outweighed by the benefit of greater accessibility.

**Comment by klonos** [comment]
Thanks for raising this and for providing a PR @herbdool 🙏🏼 

I'm not sure whether the separator should be announced or not - will need to do some research on the matter.

I agree with using a span tag, but since the character that is added as a separator may differ from language to language, I would like us to explore the possibility of adding the translated value as a `data-*` attribute to the span, and then adding that to a `:before` pseudo selector via the `content` CSS property.

**Comment by klonos** [comment]
Fair enough, but I'll have to do some more research and give it some thought, especially since the language factor is a thing. The simplest use case I can think of is RTL languages, where the » needs to be « - in some languages it may be a totally different character, so we shouldn't be "hard-coding" it.

**Comment by argiepiano** [comment]
I want to add, as a side note, that `theme_breadcrumb()` is never called in a standard installation of Backdrop. Both Seven and Basis override this theme function through `basis_breadcrumb()` and `seven_breadcrumb()` in their `template.php` file. I suspect that most themes out there will also override this theming function.

Still, I think it's a good idea to do this, since at least theme developers who choose to override it will see this and hopefully adopt a similar approach if needed.

**Comment by bugfolder** [comment]
Tested this by making Bartik the admin theme, then visiting `/admin/content`. HTML for the breadcrumb is now

```
<ol><li><a href="/">Home</a><span class="breadcrumb-separator" aria-hidden="true"> » </span></li><li><a href="/admin">Administration</a></li></ol>
```

I was wondering whether the whitespace should be outside of the span...but then I can imagine that someone might want to implement their own separator character with no whitespace by targeting the `breacrumb-separator` class. So this seems appropriate. WFM, LGTM.

**Comment by quicksketch** [comment]
Thanks @bugfolder for pushing this one over the review threshold. I merged https://github.com/backdrop/backdrop/pull/4386 into 1.x and 1.26.x. Thanks @herbdool, @klonos, and @argiepiano for working on this!
